test/test_testing.py

```diff
@@ -123,6 +123,95 @@ def test_update_relation_exposes_new_data(self):
         self.assertEqual(viewer.changes, [{'initial': 'data'},
                                           {'initial': 'data', 'new': 'value'}])
 
+    def test_update_relation_no_local_unit_change_event(self):
+        # language=YAML
+        harness = Harness(CharmBase, meta='''
+            name: my-charm
+            requires:
+              db:
+                interface: pgsql
+            ''')
+        harness.begin()
+        helper = DBRelationChangedHelper(harness.charm, "helper")
+        rel_id = harness.add_relation('db', 'postgresql')
+        harness.update_relation_data(rel_id, 'my-charm/0', {'new': 'value'})
+        rel = harness.charm.model.get_relation('db')
+        # Our unit data bag got updated.
+        self.assertTrue(rel.data[harness.charm.model.unit]['new'], 'value')
+        # But there were no changed events registered by our unit.
+        self.assertTrue(len(helper.changes) == 0)
+
+    def test_update_peer_relation_no_local_unit_change_event(self):
+        # language=YAML
+        harness = Harness(CharmBase, meta='''
+            name: postgresql
+            peers:
+              db:
+                interface: pgsql
+            ''')
+        harness.begin()
+        helper = DBRelationChangedHelper(harness.charm, "helper")
+        rel_id = harness.add_relation('db', 'postgresql')
+        harness.update_relation_data(rel_id, 'postgresql/0', {'k1': 'v1'})
+        rel = harness.charm.model.get_relation('db')
+        # Our unit data bag got updated.
+        self.assertTrue(rel.data[harness.charm.model.unit]['k1'], 'v1')
+        # But there were no changed events registered by our unit.
+        self.assertTrue(len(helper.changes) == 0)
+
+        # Same for when our unit is a leader.
+        harness.set_leader(is_leader=True)
+        harness.update_relation_data(rel_id, 'postgresql/0', {'k2': 'v2'})
+        self.assertTrue(rel.data[harness.charm.model.unit]['k2'], 'v2')
```

Comment:
I think you want assertEqual here?

Reply:
Right, failed to notice this one.

Reply:
https://github.com/canonical/operator/pull/212/commits/516c13393c0c4d4222aa2d85f520d0e37ecb0860